### DISCUSSION TITLE
Zero result return when the street in OSM do not have the street number.

### DISCUSSION THREAD

**Discussion by BobSquarePants** [osm-search/Nominatim#3441] [Q&A]
Hi everyone,

I'm currently using the API `/search?` **Structured** method to validate addresses.

in my form I have a mandatory field `house number` but it seem when OSM do not have the house number I get zero result `TheRequest.json() == []`

But when I try the same here https://nominatim.openstreetmap.org/ui/search.html?street=Rue+de+l%27%C3%89glise+2488&city=Val-David&country=Canada&postalcode=J0T+2N0

I get an result !? any ideas ?

Thanks.

**Comment by mtmail** [comment]
> in my form I have a mandatory field house number but it seem when OSM do not have the house number I get zero result 

What is the full Nominatim API URL that is called here?

> But when I try the same here https://nominatim.openstreetmap.org/ui/search.html?street=Rue+de+l%27%C3%89glise+2488&city=Val-David&country=Canada&postalcode=J0T+2N0
> I get an result

The result is a street. I'd say most users would prefer to get a nearby relevant result over getting no result.

What is the usecase for validating addresses? I mean OSM data is not complete, and not authoritive. OSM data not containing a house number doesn't mean such a house doesn't exist in the real world. (Google Maps and HERE don't know the precise location of the house number either, they use interpolation based on nearby house numbers).

**Reply by BobSquarePants** [reply]
Hi @mtmail ,

> I'd say most users would prefer to get a nearby relevant result over getting no result.

exactly ! but I received a no result ! (when called from my Python back-end, not from My Browser trough https://nominatim.openstreetmap.org/ui/search.html?... )

In my case no need to validate precisely a house number with nominatim, but be sure a street exist only, and get it's position lat, lon

**Reply by mtmail** [reply]
Can you share the relevant code that made the API request? We know nothing about your Python back-end and I don't want to make too many guesses.

**Comment by BobSquarePants** [comment]
I've so much updated my code since then, that I'm not facing this issues anymore, so I'm closing this.
Should I encounter it again in another project I'll open a new one.
Cheers, and thanks for this great piece of software !
